mfr/extensions/tabular/utilities.py

```diff
@@ -128,42 +127,52 @@ def to_bytes(fp):
     return data
 
 
+def _extract_rows(fields, raw_rows):
+    rows = []
+    for row in raw_rows:
+        if len(rows) >= MAX_SIZE:
+            break
+        rows.append(dict(zip(fields, row)))
+    return rows
+
+
 def parse_xls(wb, sheets):
     for sheet in wb.sheets():
-        verify_size(sheet.nrows, sheet.ncols, '.xls')
-        fields = fix_headers(sheet.row_values(0))
-        rows = [
-            dict(zip(fields, row_vals(sheet.row(r), wb.datemode)))
+        if getattr(sheet, 'nrows', None) is None or getattr(sheet, 'ncols', None) is None:
+            raise CorruptedError
+
+        ncols = sheet.ncols
+        max_cols = min(ncols, MAX_SIZE)
+        fields = fix_headers(sheet.row_values(0)[:max_cols])
+        raw_rows = (
+            row_vals(sheet.row(r)[:max_cols], wb.datemode)
             for r in range(1, sheet.nrows)
-        ]
+        )
+        rows = _extract_rows(fields, raw_rows)
         sheets[sheet.name] = (header_population(fields), rows)
     return sheets
 
 
 def parse_xlsx(wb, sheets):
     for name in wb.sheetnames:
         ws = wb[name]
+
+        if getattr(ws, 'max_row', None) is None or getattr(ws, 'max_column', None) is None:
+            raise CorruptedError
+
+        ncols = getattr(ws, "max_column", 0) or 0
+        max_cols = min(ncols, MAX_SIZE)
         header_row = next(ws.iter_rows(max_row=1, values_only=True), [])
-        fields = fix_headers(header_row)
-        rows = [
-            dict(zip(fields, row))
-            for row in ws.iter_rows(min_row=2,
-                                    max_row=MAX_SIZE,
-                                    max_col=MAX_SIZE,
-                                    values_only=True)
-        ]
+        fields = fix_headers(list(header_row)[:max_cols])
```

Comment:
So we discard the columns after `max_cols`? Is this the same behavior before upgrade? If so we need to inform product of this so that they can have support doc updates and/or they may ask us to put up an helper icon explaining this.